sql/src/main/java/io/crate/rest/action/RestSQLAction.java

```diff
@@ -114,10 +111,19 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         return Option.NONE;
     }
 
+    @VisibleForTesting
+    String userFromRequest(RestRequest request) {
```

Comment:
so you would rather pass the settings as argument than using the settings of the class?